### PR TITLE
Documenting cluster routing

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 * Added documentation for cluster routing to `README.md` of `api-gateway`
 * Added frontend documentation to `README.md` of `save-frontend`

### CODE REVIEW

**Review by akuleshov7 — APPROVED** [review]

File: api-gateway/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+It forwards requests to specific backend services based on their paths
+and performs some necessary filtering and header manipulation to ensure proper communication between the gateway and downstream services.
+The routes are defined based on specific URI paths:
+ - `/sec/**` is not forwarded anywhere but `api-gateway`'s `SecurityInfoController`,
+    which is responsible for `/sec/user` and `/sec/oauth-providers` endpoints
+ - `/api/sandbox` is forwarded to `save-sandbox`
+ - `/api/demo` is forwarded to `save-demo`
+ - `/api/cpg` is forwarded to `save-cpg-demo`
+ - `/api/**` is forwarded to `save-backend`
+ - `/neo4j/browser/**` is forwarded to `neo4j-browser`
```
Comment (nulls) [inline api-gateway/README.md:32]: maybe move it to ingress layer? @akuleshov7 
why api-gateway routes it. it's a separate service as grafana

File: api-gateway/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ - all the remaining requests (including resources requests) are forwarded to `save-frontend`
+
+### 3. `save-frontend`
+`save-frontend` is a pod that has `nginx` running in it (configured in `nginx.conf`).
+
+An entry point to scripts built from `save-frontend` module is `index.html`,
+so it is required to return it whenever user requests something but not resource. 
+
+After `save-frontend` pod serves `index.html`, scripts are fetched and the remaining routing is done with `basicRouting` functional component.
```
Comment (nulls) [inline api-gateway/README.md:41]: @sanyavertolet I saw you removed
```html
<!-- Kotlin/JS compiled script -->
<script src="/save-frontend.js?built-at=<%= Date.now() %>"></script>
<script src="/save-frontend-vendors.js?built-at=<%= Date.now() %>"></script>
```

Is now the page refreshed when a new frontend deployed? (I mean the problem with cookies after redeploying)

File: save-frontend/README.md [hunk 3]
```diff
@@ -7,4 +7,149 @@ To run frontend locally, use `./gradlew :save-frontend:browserDevelopmentRun --c
...
+   This directive sets the default file to serve when a **directory** is requested.
+   In this case, if a request is made to a directory (e.g., http://example.com/), Nginx will try to serve the `index.html`
+   file from that directory. If the file doesn't exist, Nginx will move to the next directive.
+2. `try_files $uri $uri/ /index.html;`
+   This directive specifies a series of files and locations that Nginx should try to serve.
+   It acts as a fallback mechanism when the requested file is not found in the specified locations.
+- `$uri` represents the requested URI.
+- `$uri/` represents the URI followed by a trailing slash, typically used for directories.
+- `/index.html` is the last fallback option.
+3. `add_header Cache-Control "private; no-store";`
```
Comment (nulls) [inline save-frontend/README.md:40]: is it done by purposes that we disable all caching?

File: save-frontend/README.md [hunk 4]
```diff
@@ -7,4 +7,149 @@ To run frontend locally, use `./gradlew :save-frontend:browserDevelopmentRun --c
...
+`setupMiddlewares` sets stubs for `/sec/user` and `/sec/oauth-providers` endpoints.
+
+`historyApiFallback` makes `webpack-dev-server` return `index.html` in case of `404 Not Found`.
+
+Sometimes it is useful to add authorization headers when proxying to some Spring services e.g. `save-backend`.
+It can be done with setting `onPorxyReq`:
+```javascript
+onProxyReq: (proxyReq, req, res) => {
+  proxyReq.setHeader("Authorization", "Basic YWRtaW46");
+  proxyReq.setHeader("X-Authorization-Source", "basic");
```
Comment (nulls) [inline save-frontend/README.md:117]: it's not used anymore (already in master)

File: save-frontend/README.md (hunk 3 above)
Comment (sanyavertolet) [reply save-frontend/README.md:40]: I guess yes, I guess, Peter tried to fix the caching issue with that

File: api-gateway/README.md (hunk 2 above)
Comment (sanyavertolet) [reply api-gateway/README.md:41]: I believe the issue with cookies is not only our issue. Haven't tested yet, but it seems that the issue remains

File: api-gateway/README.md (hunk 1 above)
Comment (sanyavertolet) [reply api-gateway/README.md:32]: Good point, will do it

File: save-frontend/README.md (hunk 4 above)
Comment (sanyavertolet) [reply save-frontend/README.md:117]: What do you mean? Did someone merge the config for `gateway` local run or what?

Comment (sanyavertolet) [reply save-frontend/README.md:117]: Or was it you who changed the behavior?

Comment (sanyavertolet) [reply save-frontend/README.md:117]: If you changed it, update the docs please 🥺🥺

Comment (nulls) [reply save-frontend/README.md:117]: `X-Authorization-Source` was removed

> If you changed it, update the docs please 🥺🥺

Will do

File: api-gateway/README.md (hunk 1 above)
Comment (nulls) [reply api-gateway/README.md:32]: neo4j uses own authentications (like grafana)

Comment (nulls) [reply api-gateway/README.md:32]: actually, neo4j browser is just browser and it does connect to neo4j database using credentials

gateway pass trough all requests to neo4j browser without authorization

Comment (nulls) [reply api-gateway/README.md:32]: raised https://github.com/saveourtool/save-cloud/issues/2406

File: save-frontend/README.md (hunk 4 above)
Comment (nulls) [reply save-frontend/README.md:117]: will do in scope https://github.com/saveourtool/save-cloud/issues/2407

File: api-gateway/README.md (hunk 1 above)
Comment (sanyavertolet) [reply api-gateway/README.md:32]: Will do it, thanks

**Pull request merged**